content/tutorials/terrain_and_DEMs/GRASS_terrain.qmd

```diff
@@ -0,0 +1,1085 @@
+
+3.  Enter 10,000 for "Minimum size of exterior watershed basin".  
+:::
+
+::: g-col-6
+![](img_terrain/r.watershed1.webp)  
+:::
+:::::
+
+::::: grid
+::: g-col-6
+4.  Under the Outputs tab, enter:  
+
+-  watershed_accumulation10000 for "accumulation",  
+
+-  watershed_basin10000 for "basins", and  
+
+-  watershed_streams10000 for "stream segments".  
+:::
+
+::: g-col-6
+![](img_terrain/r.watershed2.webp)
+:::
+:::::
+
+::::: grid
+::: g-col-6
+5.  Under the Optional tab:  
+
+-  Check the "Positive flow accumulation" box. Otherwise flow accumulation will be negative for basins at the map edge, to indicate that they may under-represent accumulation.  
+
+-  Check the "Beautify flat areas" box to have more accurate stream segments across flat areas.  
+
+-  Modify available memory as appropriate for your system.  
+
+6.  Click Run  
+:::
+
+::: g-col-6
+![](img_terrain/r.watershed3.webp)  
+:::
+:::::
+
+#### Command line
+
+```{python}
+r.watershed -a -b elevation=elevation threshold=10000 accumulation=watershed_accumulation10000 basin=watershed_basins10000 stream=watershed_streams10000
+```
+
+#### Python
+
+```{python}
+gs.run_command("r.watershed", 
+                elevation="elevation\@TerrainAnalysis", 
+                threshold="10000", 
+                accumulation=watershed_accumulation10000, 
+                basin="watershed_basin10000", 
+                stream="watershed_streams10000", 
+                flags="ab")
+
+```
+
+:::::::::::::::
+
+Resulting map of watershed basins.  
+
+![watershed basins with ≥ 10,000 cells coloring a relief map](img_terrain/watershed_basins.webp)  
+
+## Flow accumulation  
+
+Above, we also made a flow accumulation map. A flow accumulation map models the potential flow of water over a terrain:  
+
+::::: grid
+::: g-col-6
+
+-  Assuming a centimeter of rain falls on each cell of a DEM, the values of a flow accumulation map will show the depth of water in centimeters that accumulates and flows over each cell as the water flows into the cell from adjacent upslope cells.  
+
+-  The map we made assumes that the same amount of rain fell on every cell and topographic relief is the only thing affecting overland flow of water. Hence, the flow accumulation model generated represents maximal values.  
+
+-  Additional settings in *r.watershed* enable modeling overland flow more realistically with different amounts of rain falling across the region, and vegetation, soil characteristics, or barriers like dams or dikes also affecting accumulation.  
+
+-  [r.watershed](https://grass.osgeo.org/grass84/manuals/r.watershed.html) uses a **multiple flow direction** algorithm by default in which water flowing out of a cell is distributed among all adjacent downslope cells. The greatest flow is directed to the adjacent downslope cell with the greatest elevation difference and lower flows are allocated to downslope cells with lesser elevation difference.  
```

Comment:
It may be relevant to point here to the fact that the algortihm doesn't need to have sinks filled first, but it uses LCP to route water out of sinks.

Reply:
Just saw this (was hidden). Will add.